src/Compiler/Service/ServiceParsedInputOps.fs

```diff
@@ -972,11 +984,60 @@ module ParsedInput =
                                     Some (CompletionContext.ParameterList args)
                                 | _ -> 
                                     defaultTraverse expr
+
+                            | SynExpr.Match (clauses = clauses)
+                            | SynExpr.MatchBang (clauses = clauses) ->
+                                let rec traverse pos clausePat =
+                                    match clausePat with
+                                    // match x with
+                                    // | z| ->
+                                    | SynPat.Named (range = range) -> Some (CompletionContext.Match (MatchContext.ClausePatternIdentifier range))
```

Comment:
Format as 
```
                                    | SynPat.Named (range = range) ->
                                        Some (CompletionContext.Match (MatchContext.ClausePatternIdentifier range))
```

That is, if one clause goes multi-line, then make all clauses multi-line. We're gradually adopting that throughout the compiler